### PR TITLE
Fixing the generator

### PR DESCRIPTION
**Pull request by mithro**
Depends on:
- https://github.com/web-animations/web-animations-js/pull/305
- https://github.com/web-animations/web-animations-js/pull/271

### CODE REVIEW

**Comment by steveblock** [comment]
I agree that this less intrusive approach looks better

lgtm

File: test/test-generator.html [hunk 1]
```diff
@@ -184,20 +192,36 @@
...
 function printCheck(object, number, properties, time){
-  var newCheck = "check(document.querySelectorAll('" + object + "')[" + number +
-      "]," + properties + "," + time + ");";
+  var newCheck = "  at(" + time + ", function() {\n" +
+     '    assert_styles("' + object + '", [\n'
+
+  // If all the properties are the same for every object, just pass the list.
+  var allsame = properties.every(function(x) { return x === properties[0] });
+  if (allsame) {
+    newCheck = newCheck.substring(0, newCheck.length-2);
```
Comment (alancutter) [inline test/test-generator.html:201]: You could avoid this string truncation by adding the "[\n" in the else instead. This would be cleaner IMO.

Comment (mithro) [inline test/test-generator.html:201]: Fixed!

File: test/bootstrap.js [hunk 2]
```diff
@@ -83,22 +77,26 @@ function instrument(src) {
   var inst = window.__resources__[src] = new Instrumenter().instrumentSync(js, src);
 }
 
+// Can't use the setting framework to get this information because we are not
+// going to load any testharness stuff!
+if (/testharness=disable/.test(window.location.href)) {
+  loadScript('../../web-animations.js');
```
Comment (alancutter) [inline test/bootstrap.js:83]: I think you should be loading ../testharness_stub.js here.

Comment (mithro) [inline test/bootstrap.js:83]: Fixed!

**Comment by alancutter** [comment]
Some of our tests use scheduled events to check non-style values or alter the animation model, with the test harness disabled these actions are hard to time right and generate results for. This patch doesn't cover these cases.

As much as I'd like to have the generator working again I think this may be the wrong solution and merging it now will make fixing the issue more difficult than it needs to be. The generator should be using the testharness to control time just like the regular tests do.

**Comment by mithro** [comment]
This is what my first attempt at fixing the generate did, see
https://github.com/mithro/web-animations-js/tree/fixing-generator

Maybe we need to merge the two methods?

**Comment by mithro** [comment]
Will refactor and resubmit as a pull request...

**Pull request closed without merging**